rocks-lib/src/rockspec/mod.rs

```diff
@@ -46,26 +47,117 @@ pub enum RockspecError {
 #[derive(Clone, Debug)]
 pub struct Rockspec {
     /// The file format version. Example: "1.0"
-    pub rockspec_format: Option<RockspecFormat>,
+    rockspec_format: Option<RockspecFormat>,
     /// The name of the package. Example: "luasocket"
-    pub package: PackageName,
+    package: PackageName,
     /// The version of the package, plus a suffix indicating the revision of the rockspec. Example: "2.0.1-1"
-    pub version: PackageVersion,
-    pub description: RockDescription,
-    pub supported_platforms: PlatformSupport,
-    pub dependencies: PerPlatform<Vec<PackageReq>>,
-    pub build_dependencies: PerPlatform<Vec<PackageReq>>,
-    pub external_dependencies: PerPlatform<HashMap<String, ExternalDependencySpec>>,
-    pub test_dependencies: PerPlatform<Vec<PackageReq>>,
-    pub source: PerPlatform<RockSource>,
-    pub build: PerPlatform<BuildSpec>,
-    pub test: PerPlatform<TestSpec>,
+    version: PackageVersion,
+    description: RockDescription,
+    supported_platforms: PlatformSupport,
+    dependencies: PerPlatform<Vec<PackageReq>>,
+    build_dependencies: PerPlatform<Vec<PackageReq>>,
+    external_dependencies: PerPlatform<HashMap<String, ExternalDependencySpec>>,
+    test_dependencies: PerPlatform<Vec<PackageReq>>,
+    source: PerPlatform<RockSource>,
+    build: PerPlatform<BuildSpec>,
+    test: PerPlatform<TestSpec>,
     /// The original content of this rockspec, needed by luarocks
-    pub raw_content: String,
+    raw_content: String,
     /// The sha256 of this rockspec
     hash: Integrity,
 }
 
+/// A trait for querying information about a project from either a rockspec or `rocks.toml` file.
+pub trait Rocks {
```

Comment:
suggestion: I find `Rocks` a bit confusing to read, as it doesn't convey much information about what it is.
How about we name the trait `RockSpec` and rename `Rockspec` to `LuaRockSpec`?